be/src/vec/data_types/data_type_array.cpp

```diff
@@ -210,13 +217,9 @@ Status DataTypeArray::from_string(ReadBuffer& rb, IColumn* column) const {
 
         // dispose the case of [123,,,]
         if (nested_str_len == 0) {
-            if (nested_column.is_nullable()) {
-                auto& nested_null_col = reinterpret_cast<ColumnNullable&>(nested_column);
-                nested_null_col.get_nested_column().insert_default();
-                nested_null_col.get_null_map_data().push_back(0);
-            } else {
-                nested_column.insert_default();
-            }
+            auto& nested_null_col = reinterpret_cast<ColumnNullable&>(nested_column);
+            nested_null_col.get_nested_column().insert_default();
+            nested_null_col.get_null_map_data().push_back(0);
```

Comment:
We should check whether the nested column is nullable here. In future, we may enable the `not_null` syntax to create an array with the elements which are all not nullable.

Reply:
Ok. [data_type_array.cpp:185](https://github.com/apache/doris/blob/4c6cd60a020e334f4a361fecc7b9101efde83845/be/src/vec/data_types/data_type_array.cpp#L185) has `DCHECK`.

Reply:
Currently Doris only support "nullable" element inside array.
I will add a DCHECK here.
After we support "not_null" element, I will rethink this logic